File: app/src/main/java/ua/in/quireg/anothermovieapp/common/MovieReview.java
package ua.in.quireg.anothermovieapp.common;


import android.content.ContentValues;
import android.database.Cursor;

import org.json.JSONObject;

import ua.in.quireg.anothermovieapp.managers.MovieDatabaseContract;

public class MovieReview {
    public String movieId;
    public String reviewId;
    public String author;
    public String content;

    public MovieReview(String movieId, String reviewId, String author, String content) {
        this.movieId = movieId;
        this.reviewId = reviewId;
        this.author = author;
        this.content = content;
    }



    public static ContentValues contentValuesFromObj(MovieReview obj) {
        ContentValues contentValues = new ContentValues();
        contentValues.put(MovieDatabaseContract.MovieReviews._ID, obj.movieId);
        contentValues.put(MovieDatabaseContract.MovieReviews.COLUMN_REVIEW_ID, obj.reviewId);
        contentValues.put(MovieDatabaseContract.MovieReviews.COLUMN_AUTHOR, obj.author);
        contentValues.put(MovieDatabaseContract.MovieReviews.COLUMN_CONTENT, obj.content);
        return contentValues;
    }

    public static final String JSON_ID = "id";
    public static final String JSON_PAGE = "page";
    public static final String JSON_ARRAY_RESULTS = "results";
    public static final String JSON_REVIEW_ID = "id";
    public static final String JSON_AUTHOR = "author";
    public static final String JSON_CONTENT = "content";
    public static final String JSON_TOTAL_PAGES = "total_pages";

    public static MovieReview fromJSON(JSONObject obj, String movieId) {
        return new MovieReview(
                movieId,
                obj.optString(JSON_REVIEW_ID),
                obj.optString(JSON_AUTHOR),
                obj.optString(JSON_CONTENT)

        );
    }
    public static final String[] MOVIES_REVIEWS_CLOMUNS = {
            MovieDatabaseContract.MovieReviews.TABLE_NAME + "." + MovieDatabaseContract.MovieReviews._ID,
            MovieDatabaseContract.MovieReviews.COLUMN_REVIEW_ID,
            MovieDatabaseContract.MovieReviews.COLUMN_AUTHOR,
            MovieDatabaseContract.MovieReviews.COLUMN_CONTENT,

    };

    static final int COL_ID = 0;
    static final int COL_REVIEW_ID = 1;
    static final int COL_AUTHOR = 2;
    static final int COL_CONTENT = 3;

    public static MovieReview fromCursor(Cursor cursor) {
        return new MovieReview(
                cursor.getString(COL_ID),
                cursor.getString(COL_REVIEW_ID),
                cursor.getString(COL_AUTHOR),
                cursor.getString(COL_CONTENT)
        );
    }
}

File: app/src/main/java/com/anothermovieapp/repository/MovieReview.kt
package com.anothermovieapp.repository

import android.content.ContentValues
import android.database.Cursor
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.anothermovieapp.managers.MovieDatabaseContract.MovieReviews
import org.json.JSONObject

@Entity(primaryKeys = ["_id", ""], tableName = "reviews")
data class MovieReview(@PrimaryKey @ColumnInfo(name = "id") var movieId: String?,
                       @ColumnInfo(name = "reviewId") var reviewId: String,
                       @ColumnInfo(name = "author") var author: String,
                       @ColumnInfo(name = "content") var content: String) {

    companion object {
        fun contentValuesFromObj(obj: MovieReview): ContentValues {
            val contentValues = ContentValues()
            contentValues.put(MovieReviews._ID, obj.movieId)
            contentValues.put(MovieReviews.COLUMN_REVIEW_ID, obj.reviewId)
            contentValues.put(MovieReviews.COLUMN_AUTHOR, obj.author)
            contentValues.put(MovieReviews.COLUMN_CONTENT, obj.content)
            return contentValues
        }

        const val JSON_ID = "id"
        const val JSON_PAGE = "page"
        const val JSON_ARRAY_RESULTS = "results"
        const val JSON_REVIEW_ID = "id"
        const val JSON_AUTHOR = "author"
        const val JSON_CONTENT = "content"
        const val JSON_TOTAL_PAGES = "total_pages"
        fun fromJSON(obj: JSONObject, movieId: String?): MovieReview {
            return MovieReview(
                    movieId,
                    obj.optString(JSON_REVIEW_ID),
                    obj.optString(JSON_AUTHOR),
                    obj.optString(JSON_CONTENT)
            )
        }

        val MOVIES_REVIEWS_CLOMUNS = arrayOf<String?>(
                MovieReviews.TABLE_NAME + "." + MovieReviews._ID,
                MovieReviews.COLUMN_REVIEW_ID,
                MovieReviews.COLUMN_AUTHOR,
                MovieReviews.COLUMN_CONTENT)
        const val COL_ID = 0
        const val COL_REVIEW_ID = 1
        const val COL_AUTHOR = 2
        const val COL_CONTENT = 3
        fun fromCursor(cursor: Cursor?): MovieReview {
            return MovieReview(
                    cursor!!.getString(COL_ID),
                    cursor.getString(COL_REVIEW_ID),
                    cursor.getString(COL_AUTHOR),
                    cursor.getString(COL_CONTENT)
            )
        }
    }

}